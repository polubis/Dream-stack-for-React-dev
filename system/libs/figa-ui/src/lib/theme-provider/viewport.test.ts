import {
  getDownViewport,
  getUpViewport,
  isSmUp,
  isSmDown,
  VIEWPORT,
} from './viewport';

describe('Creates viewport when: ', () => {
  it('up media query is created based on given value', () => {
    expect(getUpViewport(100)).toBe('(min-width: 100px)');
  });

  it('down media query is created with one pixel less', () => {
    expect(getDownViewport(100)).toBe('(max-width: 99px)');
  });

  it('detects viewports with given functions', () => {
    expect(isSmUp(421)).toBe(true);
    expect(isSmUp(420)).toBe(true);
    expect(isSmUp(419)).toBe(false);

    expect(isSmDown(421)).toBe(false);
    expect(isSmDown(420)).toBe(false);
    expect(isSmDown(419)).toBe(true);
  });

  it('viewport object have supported values', () => {
    expect(VIEWPORT).toMatchSnapshot();
  });
});
