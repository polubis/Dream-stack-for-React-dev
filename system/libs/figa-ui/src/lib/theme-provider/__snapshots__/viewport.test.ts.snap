// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates viewport when:  viewport object have supported values 1`] = `
Object {
  "desktop": 1920,
  "laptop": 1440,
  "mobile": 600,
  "smallMobile": 420,
  "tablet": 900,
}
`;
